tests/test_encoding.py

```diff
@@ -162,6 +162,7 @@ def test_encode_linear(self):
         )
 
         param1._resolve_odxlinks(odxlinks)
+        param1._resolve_snrefs(None)
```

Comment:
Why?

Reply:
basically because of this: https://github.com/mercedes-benz/odxtools/pull/177/files/a4db1e5bb5af9caf07e75a4ff88190657e69a6e3#diff-a57adee126e4dd38d0a35068034b5c77a85e084fedbc394fa8418cb6330c1929R33-R40

there, conversion from a string to the internal value requires a DOP, but the DOP can be either referenced via ODXLINK or via SNREF and thus the conversion can only be done if we are sure that the DOP has been resolved (i.e., in `.resolve_snrefs()`). Before this patch the conversion was done at runtime very time the property was accessed. (drive-by fix, sorry for that...)